tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,200 @@
+			// TODO: Find an alternative to this list
+			"authorizationpolicies.security.istio.io",
+			"destinationrules.networking.istio.io",
+			"envoyfilters.networking.istio.io",
+			"gateways.networking.istio.io",
+			"istiorevisions.operator.istio.io",
+			"istios.operator.istio.io",
+			"peerauthentications.security.istio.io",
+			"proxyconfigs.networking.istio.io",
+			"requestauthentications.security.istio.io",
+			"serviceentries.networking.istio.io",
+			"sidecars.networking.istio.io",
+			"telemetries.telemetry.istio.io",
+			"virtualservices.networking.istio.io",
+			"wasmplugins.extensions.istio.io",
+			"workloadentries.networking.istio.io",
+			"workloadgroups.networking.istio.io",
+		}
+	)
+
+	Describe("installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("default helm manifest are applied", func() {
+			BeforeAll(func() {
+				if skipDeploy == "true" {
+					Skip("Skipping the deployment of the operator and the tests")
+				}
+				Success("Deploying Operator using default helm charts located in /chart folder")
+				Eventually(deployOperator).Should(Succeed(), "Operator failed to be deployed; unexpected error")
+			})
+
+			It("starts successfully", func() {
+				Eventually(kubectl.GetResourceCondition).
+					WithArguments(namespace, "deployment", deploymentName).
+					Should(ContainElement(resourceAvailable))
+				Success("Operator deployment is Available")
+
+				Expect(kubectl.GetPodPhase(namespace, "control-plane=istio-operator")).Should(Equal("Running"), "Operator failed to start; unexpected pod Phase")
+				Success("Istio-operator pod is Running")
+			})
+
+			It("deploys all the CRDs", func() {
+				Eventually(kubectl.GetCRDs).
+					Should(ContainElements(istioResources), "Istio CRDs are not present; expected list to contain all elements")
+				Success("Istio CRDs are present")
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		AfterAll(func() {
+			// Cleanup the operator at the end of the test
+			Eventually(undeployOperator).Should(Succeed(), "Operator failed to be deleted; unexpected error")
+			Success("Operator deployment is deleted")
+		})
+
+		for _, version := range istioVersions {
+			// Note: This var version is needed to avoid the closure of the loop
+			version := version
+
+			Context(fmt.Sprintf("version %s", version), func() {
+				BeforeAll(func() {
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed(), "Namespace failed to be created; unexpected error")
+
+					By("Creating Istio CR")
+					Eventually(createIstioCR).
+						WithArguments(version).
+						Should(Succeed(), "Istio CR failed to be created; unexpected error")
+					Success("Istio CR created")
+				})
+
+				AfterAll(func() {
```

Comment:
Should we move this to the end, after all the `When`s, so that the order matches the order of execution? This would also improve the describe output. Currently `By,Cleaning up` is in the wrong place:
```
Describe,Operator,927,7852,false,false,false,""
    Describe,installation,2084,3422,false,false,false,""
        When,default helm manifest are applied,2372,3418,false,false,false,""
            It,starts successfully,2744,3182,true,false,false,""
            It,deploys all the CRDs,3187,3413,true,false,false,""
    Describe,installation and unistallation of the istio resource,3425,7849,false,false,false,""
        Context,undefined,3861,7841,false,false,false,""
            By,Creating Istio CR,4069,4092,false,false,false,""
            By,Cleaning up,4298,4315,false,false,false,""
            When,the resource is created,4704,7346,false,false,false,""
                It,updates the Istio resource status to Reconcilied and Ready,4750,5487,true,false,false,""
...
```